pkg/asset/manifests/aws/zones_test.go

```diff
@@ -0,0 +1,862 @@
+
+func stubClusterID() *installconfig.ClusterID {
+	return &installconfig.ClusterID{
+		InfraID: "infra-id",
+	}
+}
+
+func stubInstallConfig() *installconfig.InstallConfig {
+	ic := &installconfig.InstallConfig{}
+	return ic
+}
+
+func stubInstallConfigType() *types.InstallConfig {
+	return &types.InstallConfig{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: types.InstallConfigVersion,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test-cluster",
+		},
+		Publish: types.ExternalPublishingStrategy,
+		Networking: &types.Networking{
+			MachineNetwork: []types.MachineNetworkEntry{
+				{
+					CIDR: *ipnet.MustParseCIDR(stubDefaultCIDR),
+				},
+			},
+		},
+	}
+}
+
+func stubInstallCOnfigPoolCompute() []types.MachinePool {
+	return []types.MachinePool{
+		{
+			Name: "worker",
+			Platform: types.MachinePoolPlatform{
+				AWS: &awstypes.MachinePool{
+					Zones: []string{"b", "c"},
+				},
+			},
+		},
+	}
+}
+
+func stubInstallCOnfigPoolControl() *types.MachinePool {
+	return &types.MachinePool{
+		Name: "master",
+		Platform: types.MachinePoolPlatform{
+			AWS: &awstypes.MachinePool{
+				Zones: []string{"a", "b"},
+			},
+		},
+	}
+}
+
+var stubAwsCluster = &capa.AWSCluster{
+	ObjectMeta: metav1.ObjectMeta{
+		Name:      "infraId",
+		Namespace: capiutils.Namespace,
+	},
+	Spec: capa.AWSClusterSpec{},
+}
+
+func tSortCapaSubnetsByID(in capa.Subnets) capa.Subnets {
+	subnetIds := []string{}
+	subnetsMap := make(map[string]capa.SubnetSpec, len(in))
+	for _, subnet := range in {
+		subnetsMap[subnet.ID] = subnet
+		subnetIds = append(subnetIds, subnet.ID)
+	}
+	sort.Strings(subnetIds)
+	out := capa.Subnets{}
+	for _, sid := range subnetIds {
+		out = append(out, subnetsMap[sid])
+	}
+	return out
+}
+
+func Test_extractZonesFromInstallConfig(t *testing.T) {
+	type args struct {
+		in *zonesInput
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    *zonesCAPI
+		wantErr bool
+	}{
+		{
+			name: "empty install config",
+			args: args{
+				in: &zonesInput{
+					InstallConfig: func() *installconfig.InstallConfig {
+						ic := stubInstallConfig()
+						ic.Config = nil
+						return ic
+					}(),
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "default zones",
+			args: args{
+				in: &zonesInput{
+					InstallConfig: func() *installconfig.InstallConfig {
+						ic := stubInstallConfig()
+						ic.Config = stubInstallConfigType()
+						return ic
+					}(),
+				},
+			},
+			want: &zonesCAPI{
+				controlPlaneZones: sets.Set[string]{},
+				computeZones:      sets.Set[string]{},
+			},
+		},
+		{
+			name: "custom zones control plane pool",
+			args: args{
+				in: &zonesInput{
+					InstallConfig: func() *installconfig.InstallConfig {
+						ic := stubInstallConfig()
+						ic.Config = &types.InstallConfig{
+							ControlPlane: stubInstallCOnfigPoolControl(),
+							Compute:      nil,
+						}
+						return ic
+					}(),
+				},
+			},
+			want: &zonesCAPI{
+				controlPlaneZones: sets.Set[string]{"a": {}, "b": {}},
```

Comment:
Any special reason to not just do:
```suggestion
				controlPlaneZones: sets.New("a", "b"),
```